src/xdg.rs

```diff
@@ -323,13 +411,20 @@ impl IconLoader {
     }
 
     /// Load image file as RGBA buffer.
-    fn load(&self, icon: &str, size: u32) -> Result<Icon, Error> {
+    fn load(&mut self, icon: &str, size: u32) -> Result<Icon, Error> {
```

Comment:
This function is currently only called by `set_scale_factor`, which reloads all icons when the scale factor has changed.

Since the `set_scale_factor` doesn't do anything unless the factor actually changed, this function should never be called with a `size` that is already cached.

So isn't the sorting rather useless? The only help it would be is if an icon was loaded multiple times by different desktop files, but even then why would you sort the list of icons and load it again rather than just loading the icon from the cache?

Reply:
How else would you do it?

Reply:
If the sorting isn't helpful, it's more efficient to just iterate over it.